### DISCUSSION TITLE
Automated Community Events Calendar

### DISCUSSION THREAD

**Discussion by FCLC** [hpc-social/hpc-social.github.io#51] [Ideas]
I (@vsoch) am updating @FCLC issue from a few years ago!

We have just created a new community events calendar:

https://hpc.social/events/

Which supports posting single events or feeds, and subscribing to the feed or adding a single event to the calendar. For a contribution, the form opens a GitHub issue (for further tweaking) and a maintainer can apply the `approve-event` label to have automation kick in to add the event to the calendar and close the issue.  Here is the testing example for HPSF:

<img width="1374" height="889" alt="image" src="https://github.com/user-attachments/assets/ee95fc10-3360-488e-a7ac-fc26dee46597" />

We provide views for a list, calendar, or cards, and each shows a details view. For this discussion, please use the space to provide feedback on testing, feature requests, or similar. Thanks!

-----------------

While discussion a PR for adding CaRCC's content onto the syndicated blog, there was a concern/question about what content should and shouldn't be on the blog.

One of the concerns was that their feed contained a fair few events, which in @vsoch and I's judgement didn't fit on the community blog.

On the other hand, CaRCC is not the only Org, Lab etc. that will have events, educational sessions and so on that may be of interest to those in and around HPC. 

I then proposed the following: 

> 
> A community calendar hosted somewhere on the main site, where various orgs can submit/send calendar invites (example) "[events@hpc.social](mailto:events@hpc.social)". Something as simple as a bash script+cronjob scrapes the cal.ics and adds to a central calendar marked as UTC for timecode. That calendar is marked as open and users can subscribe to it. Those interested can import/watch the "events.HPC.social" calendar if they want all events in their feed, or just look at the site to see what's coming up.
> 
> Easy to mute, avoids sharing user data with 1001 vendors, provides a service to the community etc. (I wasn't the one to set it up, but back in Uni my engineering society did something similar to avoid/centralize the colossal amount of emails different execs were getting from speakers/event planners etc. Kept our inboxes much cleaner).
> 
> Assuming the calendar invite comes with the event announcement information/link to the host site itself that should be fine? Most mailing list services for events have calendar invite functionality, so it's not extra load onto those wanting to work with hpc.social for event sharing 

Relevant PR: https://github.com/hpc-social/blog/pull/12

**Comment by alansill** [comment]
I think this bears discussion and thin that hpc.social does need an events calendar or listing feature. It should not just be a list of all events anywhere that have to do with HPC, in my opinion, however. 

What we lost in the pandemic and are still breaking best disease prevention practices to reestablish is the ability to interact with people. As vaccines improve and people riskily reengage in person, we are now losing some of the advantages that came with the ability to support remote attendance for events. I think hpc.social can help to keep some of this alive and help with the needs of those who may be immunocompromised or otherwise have difficulty attending events in person by requiring events that want to be posted to have at least one public session available to everyone without cost and with an associated follow-up activity in which people can participate. An example would be the DAAC 2022 workshop on Reducing HPC's Carbon Footprint that will be held next week (https://daac-general.nsfcac.org/daac-2022). 

To get started, we could just have a form modeled after the ones on the community map project page. We can create more sophisticated automation later. What do you think?

**Reply by vsoch** [reply]
There are two problems here I think:

- a community event feed: either via:
  - 1. syndication like rss (akin to the blog, which makes it harder to parse the time / dates)
  - 2. addition of a particular format file with metadata, akin to how I did for US-RSE
  - 3. another kind of calendar feed, e.g., ics which requires the event to distribute them.

All of those are slightly imperfect - for 1. we have a hard time parsing metadata for a calendar (unless we force some metadata to be there which is a hard ask), 2. is manual work and would only work for repeating events or requiring event providers to update manually, and 3. requires making a shared calendar. 

I don't think any of those solutions are great. Ideally we would have something akin to rss that is a feed. Maybe the best we could do is a calendar, and then have a form to submit events / content to it, and we just keep in the hpc.social space and share the one feed / embed into a site? That is more manual work, but it's not so bad to get a form response and add a new event every once in a while.

The second thing you bring up is the virtual friendly events, and as a remote person I also think this is hugely important. I'm wondering if we should start with some writeup for a policy / instructions for how to run a remote friendly event - and we could also add suggestions for making it accessible for many types of people.

**Reply by sramkrishna** [reply]
I would be interested to know what the community wants here. As the oneAPI community person, I do want to let the community know about what we are doing in regards to SYCL and math kernel library that kind of thing that could be of interest to most. But I also don't want to spam people who aren't interested on mastodon or in a blog.

**Reply by vsoch** [reply]
I think maybe the most appropriate feed would actually be something like Twitter or Mastodon, if we can make a bot? E.g., I made the us-rs jobs bot, and it posts to Twitter and slack. It's just a quick "here is  a cool thing you might be interested in" and then it basically goes away - someone could go back to our bot to browse the little feed or (more likely) click and write it down when they see it.

**Reply by vsoch** [reply]
Reddit is another idea.
